### PR TITLE
Started adding release notes for Rails 5.1 [ci skip]

### PR DESCRIPTION
**Pull request by prathamesh-sonpatki**
r? @kaspth 

Working on Action Pack next, but this can be merged.

### CODE REVIEW

File: guides/source/5_1_release_notes.md [hunk 1]
```diff
@@ -228,90 +228,140 @@ can generate form tags based on URLs, scopes or models.
...
+
 ### Removals
 
 ### Deprecations
 
-### Notable changes
+### Notable Changes
+
+*   Added support for `channel_prefix` to Redis and evented Redis adapters
+    in `cable.yml` to avoid name collisions when using same Redis server with
```
Comment (kaspth) [inline guides/source/5_1_release_notes.md]: `the same`

File: guides/source/5_1_release_notes.md [hunk 2]
```diff
@@ -228,90 +228,140 @@ can generate form tags based on URLs, scopes or models.
...
 ### Deprecations
 
-### Notable changes
+### Notable Changes
+
+*   Allowed setting custom content type when attachments are included
+    and body is set inline.
+    ([Pull Request](https://github.com/rails/rails/pull/27227))
+
+*   Allowed passing lambdas to the default hash.
```
Comment (kaspth) [inline guides/source/5_1_release_notes.md]: `default` is a method.

File: guides/source/5_1_release_notes.md [hunk 3]
```diff
@@ -228,90 +228,140 @@ can generate form tags based on URLs, scopes or models.
...
+    ([Commit](https://github.com/rails/rails/commit/1cec84ad2ddd843484ed40b1eb7492063ce71baf))
+
+*   Added support for parameterized invocation of mailers to share before filters and defaults
+    between different mailer actions.
+    ([Commit](https://github.com/rails/rails/commit/1cec84ad2ddd843484ed40b1eb7492063ce71baf))
+
+*   Passed the incoming arguments to the mailer action to `process.action_mailer` event under
```
Comment (kaspth) [inline guides/source/5_1_release_notes.md]: `under an`

File: guides/source/5_1_release_notes.md [hunk 4]
```diff
@@ -228,90 +228,140 @@ can generate form tags based on URLs, scopes or models.
...
 
 ### Deprecations
 
-### Notable changes
+### Notable Changes
+
+*   Skipped comments in the output of `mysqldump` command by default.
+    ([Pull Request](https://github.com/rails/rails/pull/23301))
+
+*   Fixed `ActiveRelation#count` to use Ruby's `Enumerable#count` for counting
```
Comment (kaspth) [inline guides/source/5_1_release_notes.md]: `ActiveRecord::Relation`

File: guides/source/5_1_release_notes.md [hunk 5]
```diff
@@ -228,90 +228,140 @@ can generate form tags based on URLs, scopes or models.
...
-Please refer to the [Changelog][active-support] for detailed changes.
-
 ### Removals
 
 ### Deprecations
 
-### Notable changes
+### Notable Changes
+
+*   Added `Module#delegate_missing_to` to delegate missing methods from a class to
```
Comment (kaspth) [inline guides/source/5_1_release_notes.md]: I'd rather say "to delegate `method_missing` and `respond_to_missing?` to a proxy object".

Comment (matthewd) [inline guides/source/5_1_release_notes.md]: I think that phrasing would imply we're calling the target's `method_missing`

Comment (kaspth) [inline guides/source/5_1_release_notes.md]: True, I just found `delegate missing methods` tough to parse.

Comment (prathamesh-sonpatki) [inline guides/source/5_1_release_notes.md]: Changed it to 

> Added `Module#delegate_missing_to` to delegate method calls not defined for the current object to a proxy object.

Thoughts?

**Comment by prathamesh-sonpatki** [comment]
Thanks @kaspth @matthewd for the review :) I updated.

**Comment by kaspth** [comment]
👏

**Pull request merged**